shared-libs/user-management/src/users.js

```diff
@@ -730,21 +730,25 @@ const hydrateUserSettings = (userSettings) => {
     });
 };
 
-const getUserDocsByName = (name) => Promise
-  .all([
-    db.users.get('org.couchdb.user:' + name),
-    db.medic.get('org.couchdb.user:' + name)
-  ])
-  .catch(error => {
-    if (error.status !== 404) {
-      return Promise.reject(error);
-    }
+const getUserDoc = (username, dbName) => db[dbName]
+  .get(`org.couchdb.user:${username}`)
+  .catch(err => {
+    err.db = dbName;
+    throw err;
+  });
 
-    return Promise.reject({
-      status: 404,
-      message: `Failed to find user with name [${name}].`
+const getUserDocsByName = (name) =>
+  Promise
+    .all(['users', 'medic'].map(dbName => getUserDoc(name, dbName)))
+    .catch(error => {
+      if (error.status !== 404) {
+        return Promise.reject(error);
+      }
+      return Promise.reject({
+        status: 404,
+        message: `Failed to find user with name [${name}] in the [${error.db}] database.`
```

Comment:
<3 <3 <3

For the longest time, I used the same installation of couchdb to run multiple projects on (which used different databases). However the users database was common. and often times I had discrepancies between entries. And _every time_ that happened, I had to manually check which database was missing the record. 

Thanks for the improvement!